src/main/java/com/microsoft/lst_bench/client/JDBCConnection.java

```diff
@@ -23,23 +23,37 @@
 public class JDBCConnection implements Connection {
 
   private final java.sql.Connection connection;
+  private final int max_num_retries;
 
-  public JDBCConnection(java.sql.Connection connection) {
+  public JDBCConnection(java.sql.Connection connection, int max_num_retries) {
     this.connection = connection;
+    this.max_num_retries = max_num_retries;
   }
 
   @Override
   public void execute(String sqlText) throws ClientException {
-    try (Statement s = connection.createStatement()) {
-      boolean hasResults = s.execute(sqlText);
-      if (hasResults) {
-        ResultSet rs = s.getResultSet();
-        while (rs.next()) {
-          // do nothing
+    Exception last_error = null;
+    int error_count = 0;
+
+    while (error_count < this.max_num_retries) {
+      try (Statement s = connection.createStatement()) {
+        boolean hasResults = s.execute(sqlText);
+        if (hasResults) {
+          ResultSet rs = s.getResultSet();
+          while (rs.next()) {
+            // do nothing
+          }
         }
+        return;
+
+      } catch (Exception e) {
+        last_error = e;
```

Comment:
Should we print a message (warn or debug level) in the log to make sure users can easily see this happened?

Reply:
Added a warn message that also includes the query text. Also added the error message and stack trace to the payload field in the telemetry to address Jose's comment. Can you verify that this is the right place to put the error?

Reply:
Sounds good @anjagruenheid !

Reply:
@netonjm , our telemetry store is a DuckDB database right now. Hopefully it could be easy for you to query that database and extract information about the run as needed (e.g., number of failing queries, actual errors, etc.)

Reply:
Yeah, I modified to use SQLite for the telemetry file, which is more common to use in our dotnet test scenario

Reply:
Please feel free to contribute that connector if you have time, it would be interesting to support it in OSS too.

Reply:
Sounds good I will prepare a PR on Monday